src/components/ellipsisMenu/EllipsisMenu.tsx

```diff
@@ -85,104 +68,64 @@ type EllipsisMenuProps = {
-        document.body.click();
-      } catch (error) {
-        console.debug(
-          "EllipsisMenu. Failed to trigger closing other menus",
-          error,
-        );
-      }
-    }
-  };
-
-  // This will set the boundary element for the Ellipsis menu popup
-  const dropdownMenuOptions = menuBoundary
-    ? {
-        modifiers: [
-          {
-            name: "flip",
-            options: {
-              boundary: menuBoundary,
-            },
-          },
-        ],
-      }
-    : undefined;
-
-  return (
-    <Dropdown alignRight onToggle={onToggle} className={className}>
-      <Dropdown.Toggle
-        className={cx(styles.toggle, toggleClassName)}
-        variant={variant}
-        size={size === "md" ? undefined : size}
+  boundingBoxRef,
+  menuButtonClassName,
+}) => (
+  <Menu
+    align="end"
+    direction="bottom"
+    gap={4}
+    boundingBoxRef={boundingBoxRef}
+    menuButton={
+      <MenuButton
         aria-label={ariaLabel}
+        className={cx(styles.button, menuButtonClassName)}
+        data-testid="ellipsis-menu-button"
+        onClick={(event) => {
+          event.stopPropagation();
+        }}
       >
         <FontAwesomeIcon icon={faEllipsisV} />
-      </Dropdown.Toggle>
-      <Dropdown.Menu popperConfig={dropdownMenuOptions} alignRight={alignRight}>
-        {items
-          .filter((x) => !x.hide)
-          .map((item) =>
-            item.href ? (
-              <Dropdown.Item
-                key={item.title}
-                href={item.href}
-                className={item.className}
-                disabled={item.disabled}
-                // There's a bug: the link stays active after clicking it
-                active={false}
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                {item.icon} {item.title}
-              </Dropdown.Item>
-            ) : (
-              <Dropdown.Item
-                key={item.title}
-                onClick={item.action}
-                className={item.className}
-                disabled={item.disabled}
-              >
-                {item.icon} {item.title}
-              </Dropdown.Item>
-            ),
-          )}
-      </Dropdown.Menu>
-    </Dropdown>
-  );
-};
+      </MenuButton>
+    }
+  >
+    {items
+      .filter((x) => !x.hide)
+      .map((item) =>
+        item.href ? (
+          <MenuItem
+            key={item.title}
+            href={item.href}
+            className={item.className}
+            disabled={item.disabled}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {item.icon}&nbsp;{item.title}
+          </MenuItem>
+        ) : (
+          <MenuItem
+            key={item.title}
+            onClick={item.action}
+            className={item.className}
+            disabled={item.disabled}
+          >
+            {item.icon}&nbsp;{item.title}
+          </MenuItem>
+        ),
```

Comment:
nit: A bit of code golf, but flatmap could make this into one callback, that is arguably the same in readability (other than some extra brackets).

```suggestion
      .flatMap((item) => 
        x.hide ? [] 
        : item.href ? [(
          <MenuItem
            key={item.title}
            href={item.href}
            className={item.className}
            disabled={item.disabled}
            target="_blank"
            rel="noopener noreferrer"
          >
            {item.icon}&nbsp;{item.title}
          </MenuItem>
        )] : [(
          <MenuItem
            key={item.title}
            onClick={item.action}
            className={item.className}
            disabled={item.disabled}
          >
            {item.icon}&nbsp;{item.title}
          </MenuItem>
      )]
```

Reply:
There's such a smaller number of possible items in all of our use cases that the extra loop is fine. And I would argue that the flatMap is absolutely less readable. It's something I would consider (or more likely, reduce) if the number of items was likely to be large

Reply:
I've seen it argued as more readable in some situations, but maybe that's just taste/depends on the kind of loop being done.

Reply:
I don't feel THAT strongly, but I would definitely rather see `.filter().map()` here if you asked me to choose